### ISSUE TITLE
enable_if error

### ISSUE THREAD

**Issue by agrawalparth10** [ucla-vision/xivo#25]
Hi, 
I am building xivo on MacOS Catalina. Getting the following error on running build.sh

```
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/type_traits:542:78: error: no type named 'type' in 'std::__1::enable_if<false, void>'; 'enable_if' cannot be used to disable this declaration
template <bool _Bp, class _Tp = void> using enable_if_t = typename enable_if<_Bp, _Tp>::type;
                                                                             ^~~
xivo/common/component.h:56:8: note: in instantiation of template type alias 'enable_if_t' requested here
  std::enable_if_t<!has_tangent<State>, void>
       ^
xivo/src/estimator.h:71:26: note: in instantiation of template class 'xivo::Component<xivo::Estimator, xivo::State>' requested here
class Estimator : public Component<Estimator, State> {
```

I am getting similar errors across different files. Any suggestions on how to resolve it?

Thank you!

**Comment by stephanietsuei** [comment]
I have never compiled xivo on any MacOS, but it looks like you are using a C++ compiler that doesn't support a particular language feature that we use. So far, we've only tested xivo using g++-7 and Ubuntu 16.04 and Ubuntu 18.04.

**Comment by stephanietsuei** [comment]
For the sake of remote work, we're going to fix this. So far, this is what we see:
- No error using gcc-7 (deprecated, because Ubuntu 16.04 is EOL) and gcc-8 (not deprecated, because Ubuntu 18.04 is not yet EOL) on Ubuntu 20.04
- Error using gcc-9 on Ubuntu 20.04
- Error using gcc-11 on Manjaro/Arch Linux
- Error on Mac OS Catalina 

@agrawalparth10, I'll get it working on the later versions of gcc first. Then I'll have you test on whatever version of Mac OS you're using.

**Comment by stephanietsuei** [comment]
I think this is fixed in commit 6be78260f8f0daccf3f01b17b68da271be5f7f16. Current status:
- Tested and runs on Ubuntu 20.04 using gcc-9. 
- Still works on Ubuntu 20.04 using gcc-8.
- Can't be tested on Manjaro/Arch until we address issue #27.
- No idea on Mac OS.

**Comment by agrawalparth10** [comment]
Okay, I will try to compile it today

**Comment by stephanietsuei** [comment]
Addressed in #28
